lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -0,0 +1,244 @@
+          zipCode: address[:zip]
+        }.compact
+      end
+
+      def add_invoice(post, money, credit_card, options)
+        post[:transaction] = {
+          id: options[:order_id],
+          dynamicDescriptor: options[:description],
+          timestamp: Time.now.utc.iso8601,
+          timezoneUtcOffset: options[:timezone_utc_offset],
+          amount: money,
+          currency: (options[:currency] || currency(money)),
+          responseCode: options[:response_code],
+          responseCodeSource: options[:response_code_source] || '',
+          avsResultCode: options[:avs_result_code],
+          cvvResultCode: options[:cvv_result_code],
+          cavvResultCode: options[:cavv_result_code],
+          cardNotPresent: credit_card.verification_value.blank?
+        }.compact
+      end
+
+      def add_payment_method(post, credit_card, address, options)
+        post[:paymentMethod] = {
+          holderName: credit_card.name,
+          cardType: 'CREDIT',
+          cardBrand: credit_card.brand&.upcase,
+          cardCountry: address[:country],
+          expirationMonth: credit_card.month,
+          expirationYear: credit_card.year,
+          cardBinNumber: credit_card.number[0..5],
+          cardLast4Digits: credit_card.number[-4..-1],
+          cardNumber: credit_card.number,
+          Token: false
+        }.compact
+      end
+
+      def address_names(address_name, payment_method)
+        split_names(address_name).tap do |names|
+          names[0] = payment_method&.first_name unless names[0].present?
+          names[1] = payment_method&.last_name unless names[0].present?
+        end
+      end
+
+      def phone_from(options)
+        options[:phone] || options.dig(:billing_address, :phone_number)
+      end
+
+      def access_token_valid?
+        @options[:access_token].present? && @options.fetch(:expires, 0) > DateTime.now.strftime('%Q').to_i
+      end
+
+      def fetch_access_token
+        params = { AppKey: @options[:app_key], AppSecret: @options[:app_secret] }
+        response = parse(ssl_post(url(:authenticate), params.to_json, headers))
+
+        @options[:access_token] = response[:accessToken]
+        @options[:expires] = response[:expires]
+        @options[:new_credentials] = true
+
+        Response.new(
+          response[:accessToken].present?,
+          message_from(response),
+          response,
+          test: test?,
+          error_code: response[:statusCode]
+        )
+      rescue ResponseError => e
+        raise OAuthResponseError.new(e)
+      end
+
+      def url(action, id = nil)
+        "#{test? ? test_url : live_url}#{ENDPOINTS_MAPPING[action] % id}"
+      end
+
+      def headers
+        { 'Content-Type' => 'application/json' }.tap do |headers|
+          headers['Authorization'] = "Bearer #{@options[:access_token]}" if @options[:access_token]
+        end
+      end
+
+      def parse(body)
+        JSON.parse(body).with_indifferent_access
```

Comment:
Why do you use `with_indifferent_access`, it's because you want to use `response[:symbol]` or some methods need to use strings in some cases and symbols in others.

Reply:
Yes, I wanted to use symbols on my accessors but I know that as an standar CORE access the params with strings